### DISCUSSION TITLE
How to write comments for model properties in generated OpenApi doc (using pydantic model)?

### DISCUSSION THREAD

**Discussion by FrederikNygaardSvendsen** [fastapi/fastapi#8770] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List, Optional
from pydantic import BaseModel


class Log(BaseModel):
    level: str
    message: str
    timestamp: int
    properties: dict = None

class CreateLogRequestDto(BaseModel):
    loggroup: str
    stream: Optional[str] = None # I would like to add a comment for this property (stream) in the generated OpenAPi spec
    logs: List[Log]
```

### Description

Hi Everyone!

I just started using FastApi a couple of days ago, and already have a microservice running in production. My question is regarding the auto generated OpenApi spec. I am using the pydantic BaseModel to declare a class, that represent the expected input (request). In the router function, i typehint that classtype for the input argument, and the generated OpenApi doc looks as expected.

However, i would love to be able to write some comments for some of the properties in the generated OpenApi spec for the request. Please see uploaded picture for reference.

![image](https://user-images.githubusercontent.com/13782727/127630656-e5469888-32ad-4b57-b085-040c673a4e54.png)

Is it possible to do so, and how?

Regards,
Frederik

### Operating System

macOS

### FastAPI Version

0.66.1

### Python Version

3.8.3

**Comment by panla** [comment, marked as answer]
the secrets are in `Body, Query, Header, Path`

example

```python
from typing import Optional, List

from fastapi import Body, Query, Header, Path
from pydantic import BaseModel
from pydantic.fields import Field


class PatchUserParser(BaseModel):
    cellphone: Optional[str] = Body(None, title='cellphone', description='user`s phone', min_length=7, max_length=15)
    name: Optional[str] = Body(None, title='name', description='user`s name', min_length=2, max_length=60)
```

the code in `fastapi.param_functions.py`

```python
def Body(  # noqa: N802
    default: Any,
    *,
    embed: bool = False,
    media_type: str = "application/json",
    alias: Optional[str] = None,
    title: Optional[str] = None,
    description: Optional[str] = None,
    gt: Optional[float] = None,
    ge: Optional[float] = None,
    lt: Optional[float] = None,
    le: Optional[float] = None,
    min_length: Optional[int] = None,
    max_length: Optional[int] = None,
    regex: Optional[str] = None,
    example: Any = Undefined,
    examples: Optional[Dict[str, Any]] = None,
    **extra: Any,
) -> Any:
```
